File: src/modules/role/validators/role_existing_validator.py
from src.modules.role.models import Role
from src.shared.exceptions.not_found_exception import NotFoundException


def role_existing_validator(role_id: int):

    role = Role.objects.filter(id=role_id, deleted_at=None).first()

    if role is None:
        raise NotFoundException("Perfil não encontrado!")

File: src/modules/role/validators/role-existing.validator.ts
import { prisma } from '../../../infra/database/prisma.service';
import { NotFoundException } from '../../../shared/exceptions';

export async function roleExistingValidator(roleId: number) {
  const role = await prisma.role.findFirst({
    where: {
      id: roleId,
      deletedAt: null,
    },
  });

  if (!role) throw new NotFoundException('Perfil não encontrado!');
}
